### PR TITLE
Optimized con_spec

### PR DESCRIPTION
**Pull request by daniel-widrick**
Rewrite for try catch logic

### CODE REVIEW

File: scripts/con_spec.js [hunk 1]
```diff
@@ -28,27 +28,22 @@ function (context, args) { // T2 CON_SPEC passes in {s:"string", d:"digit"}
 //
 // Ex: npc.loc{CON_SPEC:#s.user.con_spec}
 //
-
+
```
Comment (Pobega) [inline scripts/con_spec.js]: extra whitespace TY

**Comment by daniel-widrick** [comment]
Changes made as requested.

File: scripts/con_spec.js [hunk 2]
```diff
@@ -28,27 +28,22 @@ function (context, args) { // T2 CON_SPEC passes in {s:"string", d:"digit"}
...
-        || !((args.d && args.d.constructor == Number)
-        &&   (args.s && args.s.constructor == String))) {
+    try {
+        if(!args.d)
+            throw "D != #"
+        return (args.s.match(new RegExp(args.d,"g")) || []).length
+    } catch(e) {
         return {ok:false, msg:`
 Tier 2 \`DCON_SPEC\` lock scriptor
-
```
Comment (Pobega) [inline scripts/con_spec.js]: These newlines actually affect the output in Hackmud, they didn't need to be removed.

**Comment by Pobega** [comment]
Assigning back to @daniel-widrick just to restore the whitespace, then we can merge.

**Comment by Pobega** [comment]
I'm just going to merge and fix it myself.

**Comment by Pobega** [comment]
Thanks @daniel-widrick

**Pull request merged**